lib/rules/tap-consistent-assertions.js

```diff
@@ -0,0 +1,77 @@
+'use strict'
+
+const synonyms = require('./common/tap-synonyms.js')
+
+module.exports = {
+  meta: {
+    type: 'suggestion'
+  , fixable: 'code'
+  , docs: {
+      description: 'enforce consistent aliases for tap assertions'
+    , category: 'LogDNA'
+    }
+  , schema: [
+      {
+        type: 'object'
+      , properties: {
+          preferredMap: {
+            type: 'object'
+          }
+        }
+      , additionalProperties: false
+      }
+    ]
+  }
+, create(context) {
+    const config = context.options[0] || {}
+    if (!config.preferredMap || Object.keys(config.preferredMap).length === 0) {
+      return {}
+    }
+
+    const targetMethodMap = Object.keys(config.preferredMap).reduce((map, primary) => {
+      if (synonyms[primary]) {
+        map.set(primary, config.preferredMap[primary])
+        for (const alias of synonyms[primary]) {
+          map.set(alias, config.preferredMap[primary])
+        }
+      }
+      return map
+    }, new Map())
+
+    if (!targetMethodMap.size) {
+      return {}
+    }
+
+    const targetMethodSelectors = []
+    for (const method of targetMethodMap.keys()) {
+      targetMethodSelectors.push(`[callee.property.name="${method}"]`)
+    }
+
+    const selector = [
+      '[type="CallExpression"]'
+    , '[callee.type="MemberExpression"]'
+    , '[callee.computed=false]'
+    , '[callee.property.type="Identifier"]'
+    , `:matches(${targetMethodSelectors.join(', ')})`
+    , '[callee.object.type="Identifier"]'
+    , '[callee.object.name=/^t+$/]'
```

Comment:
is it possible to make this configurable?

Reply:
If we could either enforce  or fix these, that would be 👌 

We've talked about just make them all `t.`  instead of `ttttt.` But no one wants to fix it. 
And the single letter isn't everyones cup of tea...

Reply:
yep these will be auto-fixed and replace all of the non-preferred aliases with our preferred ones.

this does match for any number of `tttttt`s though, and it will only match when it's one of the tap assertions:

```js
tttt.same()     // would run for this
toot.equals() // would not
t.foo()            // would not
```

there are some more advanced AST-matching options too, like "only match when it's inside of a callee with name `fooBar`" but that's not foolproof (e.g. the `testCase` pattern we use sometimes)

Reply:
hmm. maybe something like

```javascript
FunctionDeclaration[params:first-child]=/^t+$/
```

Reply:
expanded the selector like so:

```js
test('foo', async (t) =>  {
  // matches
  t.same()
})
somethingElse('bar', async (tt) =>  {
  // matches
  tt.same()
})
...
// some other `ttt` variable, doesn't match
ttt.same()
```

`calleePattern` is also configurable now with the default of `/^t+$/`